src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -0,0 +1,145 @@
+import java.io.IOException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.squareup.okhttp.HttpUrl;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import java.util.ArrayList;
+import java.util.List;
+
+class Util {
+
+  private static final String CLIENT_SECRET_FILE = "/client_info.json";
+  private static final String API_KEY_FILE = "/api_key.txt";
+  private static final OkHttpClient client = new OkHttpClient();
+  private static final String INVALID_ACCESS_TOKEN = "INVALID";
+  private static final String EMPTY_REFRESH_TOKEN = "";
+  private static final String EMPTY_API_KEY = "";
+  private static final String EMPTY_PAGE_TOKEN = "";
+  private static final String ALL_DEVICES_ENDPOINT = "https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos";
+  private static final String DEFAULT_MAX_DEVICES = "200"; //is limited to effectively 200
+  private static final String DEFAULT_SORT_ORDER = "ASCENDING";
+  private static final String DEFAULT_PROJECTION = "FULL";
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final String API_KEY = getAPIKey(); 
+
+  public static List<ChromeOSDevice> getAllDevices(String userId) throws IOException {
+    final String refreshToken = getRefreshToken(userId);
+    File file = new File(Util.class.getResource(CLIENT_SECRET_FILE).getFile());
+    final GoogleClientSecrets clientSecrets =
+    GoogleClientSecrets.load(
+      JacksonFactory.getDefaultInstance(), new FileReader(file));
+    final String clientId = clientSecrets.getDetails().getClientId();
+    final String clientSecret = clientSecrets.getDetails().getClientSecret();
+    final String accessToken = getAccessToken(refreshToken, clientId, clientSecret);
+    ListDeviceResponse resp = getDevicesResponse(EMPTY_PAGE_TOKEN, accessToken);
+    final List<ChromeOSDevice> allDevices = new ArrayList<>(resp.getDevices());
+    while (resp.hasNextPageToken()) {
+      final String pageToken = (String) resp.getNextPageToken();
+      resp = getDevicesResponse(pageToken, accessToken);
+      allDevices.addAll(resp.getDevices());
+    }
+    return allDevices;
+  }
+
+  public static String getAPIKey() {
+    try {
+      File file = new File(Util.class.getResource(API_KEY_FILE).getFile());
+      String str = FileUtils.readFileToString(file);
+      return str;
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
+    }
+    return EMPTY_API_KEY;
```

Comment:
I think it's better to just let the exception thrown here, because if an empty key is returned, an exception will be thrown anyway when you use the empty to make requests.